### DISCUSSION TITLE
Fetch Existing Record of Serenity Module on Dropdown Value Change in New Add Form

### DISCUSSION THREAD

**Discussion by Hasan96-star** [serenity-is/Serenity#7153] [Help]
I have Serenity Erp based application where I already had a record against some dropdown value but what i want is like when customer go to add new record form and select dropdown value which is already present in existing record on selection of that value that old record data automatically fetch  into new record form and that form become edit form

**Comment by LSNicholls** [comment]
It sounds like you want to use the "clone" capability rather than the standard "new" behavior.  I'm not entirely clear about your flow -- from what you describe the user should press new and then it shouldn't act like new after they select the record to clone.  This isn't optimal, either from a programming point of view or from the user's point of view.  If you are "cloning", the user never selects "new".  They select the source record and then choose to "clone" it. (I have this behavior set up from both a button on a grid row and from a button on the edit form.)  When they choose to "clone" the source, your code decides which values you want to transfer and which you want to leave blank.

Does this sound good to you?  If so, there is a sample in the demo https://demo.serenity.is/BasicSamples/CloneableEntityDialog .
